src/components/Dialogs/Conditions/OrgUnitCondition.js

```diff
@@ -72,6 +79,111 @@ const OrgUnitCondition = ({
         />
     )
 }
+const getMetaDataId = (id) =>
+    ouIdHelper.hasLevelPrefix(id) || ouIdHelper.hasGroupPrefix(id)
+        ? ouIdHelper.removePrefix(id)
+        : id
+
+// VERSION-TOGGLE: rename to OrgUnitCondition when 42 is lowest supported version
+const OrgUnitConditionMinVersionV42 = ({
+    addMetadata,
+    addParentGraphMap,
+    condition,
+    onChange,
+    metadata,
+    parentGraphMap,
+}) => {
+    const { rootOrgUnits, currentUser } = useCachedDataQuery()
+    const selected = useMemo(() => {
+        const idsString = condition.split(':')?.[1] ?? ''
+
+        if (idsString.length === 0) {
+            return []
+        }
+
+        return idsString.split(';').map((id) => {
+            const isOuTreeItem =
+                !ouIdHelper.hasLevelPrefix(id) &&
+                !ouIdHelper.hasGroupPrefix(id) &&
+                !id.startsWith('USER_ORGUNIT')
+            const metaDataId = getMetaDataId(id)
+            const item = {
+                id,
+                // We store IDs for levels and groups in the metadata store without prefix
```

Comment:
The comment doesn't seem necessary - code is pretty self explanatory

Reply:
Yes perhaps the code is, but I still think this justifies some sort of comment, because of the following (copy-pasted from a Slack thread):
- When we query the backend, in the `dimension` query param, we need to use the IDs WITH the prefixes
- When we receive `metaData` from the backend we get plain IDs, WITHOUT the prefixes
- When we store things in the `metaData` redux store we do so using plain IDs, WITHOUT the prefixes
- When we interact with the `OrgUnitDimension` component (either providing the selected prop or when we read the arguments in the onSelected callback), we use the IDs WITH the prefixes

So it's not super straightforward 😅

Reply:
I'll leave it up to you. Just pointing out that the current comment basically just reiterates exactly what the code says. Maybe this deserves more of a global comment? I know we are stripping prefixes for the metadata elsewhere in the code. (FixedDimension.js)

Reply:
I'll just remove the comment